presto-docs/src/main/sphinx/security/hive.rst

```diff
@@ -0,0 +1,241 @@
+and will verify that the identity of the metastore matches
+``hive.metastore.principal``.
+
+Keytab files must be distributed to every node in the cluster that runs Presto.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+Hive HDFS Authentication
+------------------------
+
+In a Kerberized Hadoop cluster, Presto authenticates to HDFS using Kerberos.
+Kerberos authentication for HDFS is configured in the connector's properties
+file using the following properties:
+
+================================================== ============================================================ ==========
+Property Name                                      Description                                                  Default
+================================================== ============================================================ ==========
+``hive.hdfs.authentication.type``                  One of ``SIMPLE`` or ``KERBEROS``. When set to ``KERBEROS``, ``SIMPLE``
+                                                   the Hive connector authenticates to HDFS using Kerberos.
+                                                   When using the default value of ``SIMPLE``, Kerberos
+                                                   authentication is disabled and no other properties need to
+                                                   be configured.
+
+``hive.hdfs.presto.principal``                     The Kerberos principal that Presto will use when connecting
+                                                   to HDFS.
+
+``hive.hdfs.presto.keytab``                        The path to the keytab file that contains a key for the
+                                                   principal specified by ``hive.hdfs.presto.principal``.
+
+================================================== ============================================================ ==========
+
+.. _hive-security-simple:
+
+``SIMPLE``
+^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+
+``SIMPLE`` authentication is the default authentication mechanism for HDFS.
+When using ``SIMPLE`` authentication, Presto connects to HDFS using Hadoop's
+simple authentication mechanism. Kerberos is not used.
+
+.. _hive-security-kerberos:
+
+``KERBEROS``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.presto.principal=hdfs@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication, Presto accesses HDFS as the principal
+specified by the ``hive.hdfs.presto.principal`` property. Presto will
+authenticate this principal using the keytab specified by the
+``hive.hdfs.presto.keytab`` keytab.
+
+Keytab files must be distributed to every node in the cluster that runs Presto.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+.. _hive-security-impersonation:
+
+End User Impersonation
+======================
+
```

Comment:
I don't remember seeing any references to this section, other than that impersonation is mentioned in the Hive Security Configuration intro. Or, pardon my unfamiliarity with .rst,, but is the "ref" in the Hive Security Configuration intro a link to this section?